advanced_alchemy/repository/_sync.py

```diff
@@ -54,6 +54,9 @@ class SQLAlchemySyncRepository(Generic[ModelT]):
     model_type: type[ModelT]
     id_attribute: Any = "id"
     match_fields: list[str] | str | None = None
+    _prefer_any_clause: bool = False
+    prefer_any_clause_engines: tuple[str] | None = ("postgresql",)
```

Comment:
Is the union to `None` necessary here? Overriding sub-classes could set it to `()` if they wanted to override for some reason.

Reply:
It's not necessary.  I thought it might be more intuitive to also be able to set it to `None` to disable it.  Ideally, the user shouldn't be overriding this anyway.